src/main/archetype/ui.tests/test-module/specs/aem/forms.js

```diff
@@ -0,0 +1,147 @@
+        let templates = [
+                {path : '/conf/${appId}/settings/wcm/templates/basic-af', isBlankForm : false},
+                {path : '/conf/${appId}/settings/wcm/templates/blank-af', isBlankForm : true}
+            ],
+            getContainerSelector = (templateContentPath, container) => {
+                return `div[data-path="${templateContentPath}/${container}"]`;
+            },
+            verifyTemplate = (templatePath, isBlankForm) => {
+                let templateStructure = templatePath + '/structure/jcr:content',
+                    templateInitialContent = templatePath + '/initial/jcr:content',
+                    afContainerOverlaySelector = '#OverlayWrapper ' + getContainerSelector(templateInitialContent,'guideContainer'),
+                    afContainerOverlaySpanSelector = afContainerOverlaySelector + ' > span';
+
+                //Open template in editing mode
+                browser.url(`${config.aem.author.base_url}/editor.html${templatePath}/structure.html`);
+
+                //Verify all three containers are visible
+                expect($(getContainerSelector(templateStructure,'parsys1'))).toBeDisplayed();
+                expect($(getContainerSelector(templateStructure,'parsys2'))).toBeDisplayed();
+                expect($(getContainerSelector(templateStructure,'guideContainer'))).toBeDisplayed();
+
+                //Switch to initial content layer
+                $(selectors.editor.layerSwitcher).click();
+                expect($(selectors.editor.layerSelector.initial).waitForDisplayed()).toBe(true);
+                $(selectors.editor.layerSelector.initial).click();
+
+                //Click on specific area of AF container overlay and verify toolbar
+                expect($(afContainerOverlaySelector).waitForDisplayed()).toBe(true);
+                $(afContainerOverlaySelector).focus();
+                expect($(afContainerOverlaySpanSelector).waitForDisplayed()).toBe(true);
+                $(afContainerOverlaySpanSelector).click();
+
+                //In toolbar, click on configure button and verify properties container
+                expect($(selectors.editor.editToolBar.configure).waitForDisplayed()).toBe(true);
+                $(selectors.editor.editToolBar.configure).click();
+                expect($(selectors.editor.sidePanel.afPropertiesContainer).waitForDisplayed()).toBe(true);
+
+                //Switching back to Structure layer
+                $(selectors.editor.layerSwitcher).click();
+                expect($(selectors.editor.layerSelector.structure).waitForDisplayed()).toBe(true);
+                $(selectors.editor.layerSelector.structure).click();
+
+                //Preview the template using url and verify AF field presence in the form
+                browser.url(`${config.aem.author.base_url}/${templatePath}/initial.html?wcmmode=preview`);
+                expect($(selectors.content.genericAFField).waitForDisplayed({ reverse: isBlankForm })).toBe(true);
```

Comment:
Can we have descriptive error message for each expect ?

Reply:
@rismehta,  wdio-expect doesn't seem to support custom messages. It is using builtin Jest's expect assertions
https://webdriver.io/docs/api/expect-webdriverio/#default-matchers